src/data_source/fetching.rs

```diff
@@ -539,6 +545,39 @@ where
     }
 }
 
+#[async_trait]
```

Comment:
I think we need a similar blanket `UpdateStateData` here, like we have in `ExtensibleDataSource`. Basically anything that wraps an inner data source should blanket implement these traits, just in case the inner data source happens to implement them